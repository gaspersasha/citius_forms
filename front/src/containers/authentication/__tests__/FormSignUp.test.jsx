import React from 'react';
import { FormSignUp } from '..';

const props = {
  extendedActiveForm: jest.fn(),
  setFieldValue: jest.fn(),
  onSubmit: jest.fn(),
  isUserRemembered: true,
  isUnsubscribed: false,
  errorMessage: '',
  activeForm: 'signup',
  firstNameValue: 'FirstName',
  lastNameValue: 'LastName',
};

describe('<FormSignUp />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<FormSignUp {...props} />);
  });

  it('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setActiveForm function', () => {
    const loginButton = wrapper.find('b[data-id="link"]');

    loginButton.simulate('click');
    expect(props.extendedActiveForm).toHaveBeenCalledWith('login');
  });

  it('should call setFieldValue function', () => {
    const unsubscribeCheckbox = wrapper.find('input[type="checkbox"]');

    unsubscribeCheckbox.at(1).simulate('change');
    expect(props.setFieldValue).toHaveBeenCalledWith('unsubscribe', true);
  });
});
