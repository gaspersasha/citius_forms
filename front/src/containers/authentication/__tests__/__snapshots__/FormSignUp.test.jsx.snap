// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should match snapshots 1`] = `
<FormSignUp
  activeForm="signup"
  emailValue=""
  errorMessage=""
  extendedActiveForm={[MockFunction]}
  firstNameValue="FirstName"
  isUnsubscribed={false}
  isUserRemembered={true}
  lastNameValue="LastName"
  onSubmit={[MockFunction]}
  setFieldValue={[MockFunction]}
>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="wrapper"
    >
      <AuthForm
        activeForm="signup"
        emailValue=""
        errorMessage=""
        firstNameValue="FirstName"
        isUserRemembered={true}
        lastNameValue="LastName"
        onSubmit={[MockFunction]}
        setFieldValue={[MockFunction]}
        subTitle="Enter your email address and password"
        title="Create a BuyaCar account"
      >
        <h6
          className="typo title"
        >
          Create a BuyaCar account
        </h6>
        <span
          className="typo typo__small subTitle"
        >
          Enter your email address and password
        </span>
        <div
          className="row"
        >
          <div
            className="firstName"
          >
            <InputText
              className="typo input"
              customClasses="auth"
              errorMessage=""
              id="firstName"
              label=""
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              required={false}
              status=""
              subTitle=""
              value="FirstName"
            >
              <Label
                customClasses="auth"
                name="firstName"
                required={false}
                status=""
                title=""
                withSub={false}
              >
                <div
                  className="label auth"
                >
                  <label
                    data-id={false}
                    htmlFor="firstName"
                  >
                    <span
                      className="title"
                      data-id="label-title"
                    />
                    <input
                      className="typo input"
                      id="firstName"
                      name="firstName"
                      onChange={[Function]}
                      placeholder="First Name"
                      type="text"
                      value="FirstName"
                    />
                  </label>
                </div>
              </Label>
            </InputText>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="lastName"
          >
            <InputText
              className="typo input"
              customClasses="auth"
              errorMessage=""
              id="lastName"
              label=""
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              required={false}
              status=""
              subTitle=""
              value="LastName"
            >
              <Label
                customClasses="auth"
                name="lastName"
                required={false}
                status=""
                title=""
                withSub={false}
              >
                <div
                  className="label auth"
                >
                  <label
                    data-id={false}
                    htmlFor="lastName"
                  >
                    <span
                      className="title"
                      data-id="label-title"
                    />
                    <input
                      className="typo input"
                      id="lastName"
                      name="lastName"
                      onChange={[Function]}
                      placeholder="Last Name"
                      type="text"
                      value="LastName"
                    />
                  </label>
                </div>
              </Label>
            </InputText>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="email"
          >
            <InputEmail
              autoFocus={true}
              className="typo input"
              customClasses="auth"
              errorMessage="The email address you entered is not valid, please try again"
              id="authEmail"
              label={false}
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              required={false}
              status=""
              subTitle=""
              useErrorBorder={true}
              value=""
            >
              <Label
                customClasses="auth"
                name="email"
                required={false}
                status=""
                title={false}
                withSub={false}
              >
                <div
                  className="label auth"
                >
                  <label
                    data-id={false}
                    htmlFor="email"
                  >
                    <span
                      className="title"
                      data-id="label-title"
                    />
                    <input
                      autoFocus={true}
                      className="typo input"
                      id="authEmail"
                      name="email"
                      onChange={[Function]}
                      placeholder="Email Address"
                      type="email"
                      value=""
                    />
                  </label>
                </div>
              </Label>
            </InputEmail>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="password"
          >
            <InputPassword
              className="typo input"
              customClasses="auth"
              id="userPassword"
              label={false}
              name="password"
              onChange={[Function]}
              required={false}
              status=""
              subTitle=""
              type="password"
              value=""
            >
              <Label
                customClasses="auth"
                name="password"
                required={false}
                status=""
                title={false}
                withSub={false}
              >
                <div
                  className="label auth"
                >
                  <label
                    data-id={false}
                    htmlFor="password"
                  >
                    <span
                      className="title"
                      data-id="label-title"
                    />
                    <input
                      className="typo input"
                      data-id="password-input"
                      defaultValue=""
                      id="userPassword"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your password"
                      type="password"
                    />
                  </label>
                </div>
              </Label>
            </InputPassword>
          </div>
        </div>
        <div
          className="row row__largeBottomSpace"
        >
          <InputCheckbox
            checked={true}
            className="checkbox"
            customClasses="auth"
            id="authRememberMe"
            label="Remember me"
            name=""
            onChange={[Function]}
            status=""
          >
            <div
              className="wrapper auth"
            >
              <input
                checked={true}
                className="checkbox"
                id="authRememberMe"
                name=""
                onChange={[Function]}
                type="checkbox"
              />
              <Label
                customClasses="auth"
                name=""
                status=""
                title="Remember me"
                withSub={false}
              >
                <div
                  className="label auth"
                >
                  <label
                    data-id={false}
                    htmlFor=""
                  >
                    <span
                      className="title"
                      data-id="label-title"
                    >
                      Remember me
                    </span>
                  </label>
                </div>
              </Label>
            </div>
          </InputCheckbox>
          <span
            className="typo typo__small typo__pointer"
            data-id="toggle-pass"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            Show password
          </span>
        </div>
        <div
          className="row row__middleBottomSpace"
        >
          <Button
            className="typo submit"
            data-id="login"
            disabled={false}
            href=""
            onClick={[MockFunction]}
            selected={false}
            styleType="primary"
            type="submit"
          >
            <button
              className="buttonComponent primary typo submit"
              data-id="login"
              disabled={false}
              href={null}
              onClick={[MockFunction]}
              type="submit"
            >
              Sign Up
            </button>
          </Button>
        </div>
      </AuthForm>
      <p
        className="typo typo__small link"
      >
        <span>
          Already have a BuyaCar account? 
        </span>
        <b
          data-id="link"
          onClick={[Function]}
          role="dialog"
        >
          Login
        </b>
      </p>
      <div
        className="privacyNotice"
      >
        <p>
          Privacy Notice
          <br />
          We will use the details you have shared to manage your car purchase. You agree to the processing, storage, sharing and use of this information for the purpose of managing your car purchase as described in our
          <a
            className="privacyLinks"
            href="https://www.autovia.co.uk/privacy-policy/#vehicle-purchase-or-enquiry"
            rel="noopener noreferrer"
            target="_blank"
          >
             Privacy Policy
          </a>
          .
        </p>
        <div
          className="field_type_text__auth"
        >
          We believe that based on this purchase you may be interested in other related products and services we offer. As described in our
          <a
            className="privacyLinks"
            href="https://www.autovia.co.uk/privacy-policy/#vehicle-purchase-or-enquiry"
            rel="noopener noreferrer"
            target="_blank"
          >
             Privacy Policy 
          </a>
          we will use the information you have shared to send you communications about such products and services. If you do not wish to receive these communications from us then please click here and you will be unsubscribed from this activity.
          <InputCheckbox
            checked={false}
            className="checkbox checkbox__unsubscribe"
            customClasses="sign_up"
            label=""
            name=""
            onChange={[Function]}
            status=""
          >
            <div
              className="wrapper sign_up"
            >
              <input
                checked={false}
                className="checkbox checkbox__unsubscribe"
                name=""
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </InputCheckbox>
        </div>
        <p>
          We includes BuyaCar, AutoExpress, Carbuyer, evo and other Autovia Group brands as
          <a
            className="privacyLinks"
            href="https://www.autovia.co.uk/brands"
            rel="noopener noreferrer"
            target="_blank"
          >
             detailed here
          </a>
          .
        </p>
      </div>
    </div>
  </form>
</FormSignUp>
`;
